### DISCUSSION TITLE
Hub replaced and new api key

### DISCUSSION THREAD

**Discussion by videojedi** [CJNE/ha-myenergi#397] [Q&A]
Does anyone know where I can update the api key and serial number of the hub? I have a new hub (Libbi) and would really like to replace the api key and serial number, rather than remove/add the integration, as this will erase all the previous energy data.

Thanks

**Comment by LeiChat** [comment]
Sorry, not an answer to your question but one way to retain long term statistics for energy devices when they are replaced is to use a Utility Meter Helper for each of the key stats you wish to track.

https://www.home-assistant.io/integrations/utility_meter/

This guy explains the principle for "imported energy" but I'm presuming we could use them for storage battery import, hot water tank boost, etc.
https://www.youtube.com/watch?v=YPPpwTKIz7M

**Comment by videojedi** [comment]
I've sort of answered my question.....
There is a file that stores persistent values of things like credentials etc. /config/.storage/core.config_entries. Lots of documentation says, "don't touch this file'............. well i like to break things. :)

I enabled ssh and logged into my home assistant. I then stopped the HAcore and edited the file with the new api key and serial number. then restarted the core. Hey presto, my new hub appeared and all the previous devices remained connected.

However, in the energy dashboard, if you replace the Solar panel generation sensor, you will loose all the previous data. So instead I just added the new sensor from the new hub. I guess I could have just renamed the new hub, to match the old.

When I'm tired of reading the old data, I will just erase that sensor. But for now it can just stay as 'un-available'.
